capi/geos_c.h.in

```diff
@@ -3418,11 +3456,25 @@ extern char GEOS_DLL GEOSisValidDetail(
 
 /**
 * Repair an invalid geometry, returning a valid output.
-* \param g The geometry to test
+* \param g The geometry to repair
 * \return The repaired geometry. Caller must free with GEOSGeom_destroy().
 */
 extern GEOSGeometry GEOS_DLL *GEOSMakeValid(const GEOSGeometry* g);
 
+/**
+* Repair an invalid geometry, returning a valid output, using the
+* indicated GEOSMakeValidMethods algorithm and options.
+* \param g is the geometry to test.
+* \param makeValidMethod indicates what repair algorithm to use.
+* \param makeValidOptions to control the behaviour or the repair method.
+* \return A repaired geometry. Caller must free with GEOSGeom_destroy().
+*/
+extern GEOSGeometry GEOS_DLL *GEOSMakeValidWithOptions(
```

Comment:
I'd call this `GEOSMakeValidWithParams` for similarity to `GEOSBufferWithParams`, unless there is some distinction between "options" and "params"